### PR TITLE
revert: manual management of detached WebContents

### PR DESCRIPTION
**Pull request by alexstrat**
#### Description of Change
Following discussions in #18976, revert #14487

This [CL](https://chromium.googlesource.com/chromium/src/+/7464a6b4e3cf91dd1eb96d922736f0815b94ed9c%5E%21/#F3) that [landed with 76.0.3783.1](https://github.com/electron/electron/pull/18116) made #14487 not useful anymore.

I did follow the reproduction steps explained in #14211 and could not reproduce the issue.

cc @zcbenz @deepak1556
#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Revert electron/electron#14487

### CODE REVIEW

File: shell/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -460,7 +455,7 @@ WebContents::~WebContents() {
     } else {
       // Destroy WebContents asynchronously unless app is shutting down,
       // because destroy() might be called inside WebContents's event handler.
-      DestroyWebContents(true /* async */);
```
Comment (alexstrat) [inline shell/browser/api/atom_api_web_contents.cc]: I'm not sure `!IsGuest()` is what we expect here.

File: shell/browser/api/atom_api_web_contents.cc [hunk 2]
```diff
@@ -460,7 +455,7 @@ WebContents::~WebContents() {
...
+      DestroyWebContents(!IsGuest() /* async */);
```
Comment (alexstrat) [inline shell/browser/api/atom_api_web_contents.cc]: I'm not sure `!IsGuest()` is what we expect here. @zcbenz opinion?

**Comment by alexstrat** [comment]
@deepak1556 you were suggesting to add tests: what kind of tests would you suggest?

File: shell/browser/api/atom_api_web_contents.cc (hunk 2 above)
Comment (zcbenz) [reply shell/browser/api/atom_api_web_contents.cc]: It is correct.

**Comment by zcbenz** [comment]
> @deepak1556 you were suggesting to add tests: what kind of tests would you suggest?

I couldn't find a way to programmably test the original issue, it seemed to be only visually noticeable.

**Comment by jkleinsc** [comment]
@alexstrat can you rebase this PR?

**Comment by alexstrat** [comment]
@jkleinsc rebase done

**Comment by zcbenz** [comment]
This PR needs an approval from @electron/wg-upgrades.

**Comment by deepak1556** [comment]
@alexstrat sorry about this but can you rebase again on master to pull in https://github.com/electron/electron/pull/19127 that should fix the macOS build error. Thanks!

**Comment by alexstrat** [comment]
@deepak1556 done

**Comment by codebytere** [comment]
macOS failure is expected, as code-signing tests fail on PRs from forks. 
will merge once linux CI has passed.

**Pull request merged**